plugins/outputs/cloudwatchlogs/internal/pusher/target.go

```diff
@@ -61,11 +62,19 @@ func (m *targetManager) InitTarget(target Target) error {
 	m.mu.Lock()
 	defer m.mu.Unlock()
 	if _, ok := m.cache[target]; !ok {
-		err := m.createLogGroupAndStream(target)
+		newGroup, err := m.createLogGroupAndStream(target)
 		if err != nil {
 			return err
 		}
-		m.PutRetentionPolicy(target)
+		if target.Retention > 0 {
+			if newGroup {
+				if err := m.processWithRetry(target, true); err != nil {
```

Comment:
This makes one of the branches a blocking call and other not. Would prefer if we could offload it to the other thread and put it in the channel. No need to block sends or reads on whether the put retention policy has been set yet.